tests/dirent/test_readdir.c

```diff
@@ -130,7 +129,7 @@ void test() {
   ent = readdir(dir);
   assert(ent);
   //printf("check: %s / %s\n", ent->d_name, first);
-  assert(!strcmp(ent->d_name, first));
```

Comment:
This tests looks a little strange to me.

It seems to be reading the name of the first entry.   Then using telldir() to store the current location.  Then when seeking back to the location which should *not* be the first but the second
entry, no?   What am I missing?  How did this test ever pass at all?

Reply:
This looks like a real bug in out readdir.   

loc is being set to point to the second entry, but then checked to equal the first.   We should be setting loc before reading the entry, or alternatively (better) comparing against the second entry not the first.

Reply:
I filed a separate bug and PR to disable this part of the test: https://github.com/emscripten-core/emscripten/pull/8527

Reply:
I don't think this is a bug. It certainly looks strange and is counterintuitive when considering the semantics of seek/read in a file descriptor context. The behavior as tested in valid after doing a `rewinddir()`. I tested this on a few systems when I was implementing `readdir()` in a previous PR.

Reply:
(The reason I changed this line is that directory order is not deterministic, not because of the strangeness of `seekdir()`'s behavior after `rewinddir()`.)

Reply:
Hmm yeah I think you're right, this is a bug. Probably in the implementation of `__syscall220`